chimney/src/main/scala/io/scalaland/chimney/internal/ErrorsCollection.scala

```diff
@@ -0,0 +1,41 @@
+package io.scalaland.chimney.internal
+
+import io.scalaland.chimney.PartialTransformer
+
+sealed trait ErrorsCollection extends Iterable[PartialTransformer.Error] {
+  def iterator: Iterator[PartialTransformer.Error] = {
+    this match {
+      case _: ErrorsCollection.Empty.type => Iterator.empty
+      case ErrorsCollection.Single(error) => Iterator.single(error)
+      case ErrorsCollection.Wrap(errors) => Iterator.from(errors)
+      case ErrorsCollection.Merge(left, right) => left.iterator ++ right.iterator
+    }
+  }
+
+  def ++(other: ErrorsCollection): ErrorsCollection = ErrorsCollection.Merge(this, other)
+
+  override def equals(obj: Any): Boolean = {
+    obj match {
+      case xs: Iterable[_] => this.iterator.sameElements[Any](xs.iterator)
+      case _ => false
+    }
+  }
+}
+
+object ErrorsCollection {
+  val empty: ErrorsCollection = Empty
+
+  def fromSingle(error: PartialTransformer.Error): ErrorsCollection = Single(error)
+
+  def fromIterable(iterable: Iterable[PartialTransformer.Error]): ErrorsCollection = {
+    iterable match {
+      case ec: ErrorsCollection => ec
+      case _ => Wrap(iterable)
+    }
+  }
+
+  case object Empty extends ErrorsCollection
+  case class Single(error: PartialTransformer.Error) extends ErrorsCollection
+  case class Wrap(errors: Iterable[PartialTransformer.Error]) extends ErrorsCollection
+  case class Merge(left: ErrorsCollection, right: ErrorsCollection) extends ErrorsCollection
```

Comment:
Once we have benchmarks would be nice to check if making `right` lazy would affect performance.

Reply:
Not much difference.

```
with earer Merge.right:

[info] ErrorAccTransformersBenchmark.longHappyPartialTransformer               thrpt    5    4112.814 ±   23.314  ops/ms
[info] ErrorAccTransformersBenchmark.longUnhappyPartialTransformer             thrpt    5     939.255 ±   56.124  ops/ms
[info] ErrorAccTransformersBenchmark.nestedLongHappyPartialTransformer         thrpt    5      19.500 ±    0.110  ops/ms
[info] ErrorAccTransformersBenchmark.nestedLongUnhappyPartialTransformer       thrpt    5       3.780 ±    0.101  ops/ms
[info] ErrorAccTransformersBenchmark.simpleHappyInlineDslPartialTransformer    thrpt    5   11496.315 ± 2203.400  ops/ms
[info] ErrorAccTransformersBenchmark.simpleHappyPartialTransformer             thrpt    5   54377.346 ±  545.966  ops/ms
[info] ErrorAccTransformersBenchmark.simpleUnhappyInlineDslPartialTransformer  thrpt    5   10936.705 ±  110.292  ops/ms
[info] ErrorAccTransformersBenchmark.simpleUnhappyPartialTransformer           thrpt    5   16745.224 ±   66.072  ops/ms

with lazy Merge.right:

[info] ErrorAccTransformersBenchmark.longHappyPartialTransformer               thrpt    5    3976.027 ±   107.518  ops/ms
[info] ErrorAccTransformersBenchmark.longUnhappyPartialTransformer             thrpt    5     963.235 ±     6.112  ops/ms
[info] ErrorAccTransformersBenchmark.nestedLongHappyPartialTransformer         thrpt    5      19.327 ±     0.057  ops/ms
[info] ErrorAccTransformersBenchmark.nestedLongUnhappyPartialTransformer       thrpt    5       3.766 ±     0.020  ops/ms
[info] ErrorAccTransformersBenchmark.simpleHappyInlineDslPartialTransformer    thrpt    5   12855.889 ±  2001.966  ops/ms
[info] ErrorAccTransformersBenchmark.simpleHappyPartialTransformer             thrpt    5   53683.988 ±  1168.589  ops/ms
[info] ErrorAccTransformersBenchmark.simpleUnhappyInlineDslPartialTransformer  thrpt    5   11027.271 ±    34.923  ops/ms
[info] ErrorAccTransformersBenchmark.simpleUnhappyPartialTransformer           thrpt    5   16637.188 ±   285.618  ops/ms

```